### PR TITLE
intructions to reviewer - string-calculator

### PR DESCRIPTION
**Pull request by Sbonelo01**
Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: content/projects/tdd/string-calculator/_index.md [hunk 1]
```diff
@@ -205,4 +205,14 @@ add("1,2,3//;\n1000,1;2")
...
 
-- If you're wondering to yourself, "Why is this sooo important!?" take a look at {{% contentlink path="/topics/data_validation_and_verification/" %}}
\ No newline at end of file
+- If you're wondering to yourself, "Why is this sooo important!?" take a look at {{% contentlink path="/topics/data_validation_and_verification/" %}}
+
+## Instructions For Reviewers
+
+- Look out for file structure. It might seem obvious, but people sometimes mess up the file structure of a TDD project. Refer to : **Set up your environment** which is at the top of this page.
+
+- Code should be clean, DRY and should follow the KISS(Keep It Simple, Stupid) principal - look out for these when reviewing. Also ensure that they have used a tool like Prettier or Black to format their code.
```
Comment (ry-oc) [inline content/projects/tdd/string-calculator/_index.md]: The review instructions should be project specific.

Clean code conventions should not be mentioned in the review instructions as this relates to every project.

**Review by ry-oc — CHANGES_REQUESTED** [review]
Small change

File: content/projects/tdd/string-calculator/_index.md [hunk 2]
```diff
@@ -209,10 +209,10 @@ add("1,2,3//;\n1000,1;2")
 
 ## Instructions For Reviewers
 
-- Look out for file structure. It might seem obvious, but people sometimes mess up the file structure of a TDD project. Refer to : **Set up your environment** which is at the top of this page.
+- Tests are nessesary for this project. Proper testing to be followed. If someone uses mocks and spies - that is incompetent.
 
-- Code should be clean, DRY and should follow the KISS(Keep It Simple, Stupid) principal - look out for these when reviewing. Also ensure that they have used a tool like Prettier or Black to format their code.
+- Make sure that the output is as per project instructions. Proper error handling to be used. no `print`, `console.log`. an error is to be thrown when needed and a value returned when needed.
```
Comment (ry-oc) [inline content/projects/tdd/string-calculator/_index.md]: Remove this piece: `Make sure that the output is as per project instructions.` as I think that is a given for any project :)

File: content/projects/tdd/string-calculator/_index.md [hunk 3]
```diff
@@ -209,10 +209,10 @@ add("1,2,3//;\n1000,1;2")
...
 
-- Make sure that the output is as per project instructions. test everything.
+- Ensure that the add function takes in the correct data-type as argument - the datatype should be a string.
```
Comment (ry-oc) [inline content/projects/tdd/string-calculator/_index.md]: Could this sentence be simplified to something like: `Ensure that the add function takes a string as an argument.

File: content/projects/tdd/string-calculator/_index.md [hunk 4]
```diff
@@ -209,10 +209,10 @@ add("1,2,3//;\n1000,1;2")
 
 ## Instructions For Reviewers
 
-- Look out for file structure. It might seem obvious, but people sometimes mess up the file structure of a TDD project. Refer to : **Set up your environment** which is at the top of this page.
+- Tests are nessesary for this project. Proper testing to be followed. If someone uses mocks and spies - that is incompetent.
```
Comment (ry-oc) [inline content/projects/tdd/string-calculator/_index.md]: this part can be removed as it is in the project instructions: `Tests are nessesary for this project. `

**Review by ry-oc — CHANGES_REQUESTED** [review]
3 tiny things, please Dm me when changes are made.

**Review by ry-oc — APPROVED** [review]

**Review by ry-oc — APPROVED** [review]

**Review by sheenarbw — CHANGES_REQUESTED** [review]
- There is to be only one function. The `add()` function.

I disagree.  Making one long ass function is often bad practice. 

The `add` function needs to work. But if the learner decides to let `add` rely on other functions then that's totally fine. Often breaking a big function down into multiple smaller functions adds clarity, reusability and testability to the code.

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**